dockstore-common/src/main/java/io/dockstore/common/yaml/constraints/DescriptorLanguageShortNameValidator.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright 2022 OICR and UCSC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.dockstore.common.yaml.constraints;
+
+import io.dockstore.common.DescriptorLanguage;
+import javax.validation.ConstraintValidator;
+import javax.validation.ConstraintValidatorContext;
+
+/**
+ * Validates a descriptor language short name.
+ */
+public class DescriptorLanguageShortNameValidator implements ConstraintValidator<DescriptorLanguageShortName, String> {
+    @Override
+    public void initialize(final DescriptorLanguageShortName constraintAnnotation) {
+        // Intentionally empty
+    }
+
+    @Override
+    public boolean isValid(final String shortName, final ConstraintValidatorContext context) {
+        if (shortName == null) {
+            return true;
+        }
+        try {
+            DescriptorLanguage.convertShortStringToEnum(shortName);
+            return true;
+        } catch (UnsupportedOperationException ex) {
+            return false;
```

Comment:
Would it be useful to log `shortName`, or is that already going to be in the `ConstraintViolation`? Even if it is, it might be useful to have in the web service logs if we're trying to track down why a particular push didn't work.

Reply:
Cool thing about the validation code: when it creates the exception message, it includes the current [wrong] value.

Reply:
Oh, also, we log the exception message, so it'll be there.